src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -92,6 +92,25 @@ public static string GetDirectoryName(string path)
             return null;
         }
 
+        public static ReadOnlySpan<char> GetDirectoryName(ReadOnlySpan<char> path)
+        {
+            if (path.IsEmpty)
+                return path;
+
+            path = PathInternal.NormalizeDirectorySeparators(path);
+            int root = PathInternal.GetRootLength(new string(path));
+
+            int i = path.Length;
+            if (i > root)
+            {
+                while (i > root && !PathInternal.IsDirectorySeparator(path[--i]))
+                    ;
+                return path.Slice(0, i);
+            }
+
+            return new ReadOnlySpan<char>();
```

Comment:
change instances of the empty constructor to `ReadOnlySpan<char>.Empty` as it's slightly cleaner.